src/XRP/xrp-transaction.ts

```diff
@@ -0,0 +1,110 @@
+import { Transaction } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import RippledFlags from '../rippled-flags'
+import XRPSigner from './xrp-signer'
+import XRPTransactionType from './xrp-transaction-type'
+import XRPPayment from './xrp-payment'
+import XRPMemo from './xrp-memo'
+
+/*
+ * A transaction on the XRP Ledger.
+ *
+ * @see: https://xrpl.org/transaction-formats.html
+ */
+// TODO(amiecorso): Modify this object to use X-Address format.
+export default class XRPTransaction {
+  public static from(transaction: Transaction): XRPTransaction | undefined {
+    const account = transaction
+      .getAccount()
+      ?.getValue()
+      ?.getAddress()
+
+    const fee = transaction.getFee()?.getDrops()
+
+    const sequence = transaction.getSequence()?.getValue()
+
+    const signingPublicKey = transaction.getSigningPublicKey()?.getValue_asU8()
+
+    const transactionSignature = transaction
+      .getTransactionSignature()
+      ?.getValue_asU8()
+
+    const accountTransactionID = transaction
+      .getAccountTransactionId()
+      ?.getValue_asU8()
+
+    const flags = transaction.getFlags()?.getValue()
```

Comment:
Really like the use of the `?` operator chaining here to get a value | undefined. Nice!

Reply:
Wondering if these return types should be more explicit?

Reply:
What would that look like?  type annotations in the variables?  This is number 9/9 in a chain of classes like this so for the moment I'm going to leave as-is, since the types of these variables are essentially documented in the constructor.

Reply:
Yeah, sounds good